x-pack/plugins/secops/public/pages/hosts/index.tsx

```diff
@@ -14,57 +14,68 @@ import { EmptyPage } from '../../components/empty_page';
 import { HorizontalBarChartData } from '../../components/horizontal_bar_chart';
 import { EventsTable, HostsTable, TypesBar } from '../../components/page/hosts';
 
+import { manageQuery } from '../../components/page/manage_query';
 import { EventsQuery } from '../../containers/events';
+import { GlobalTime } from '../../containers/global_time';
 import { HostsQuery } from '../../containers/hosts';
 import { WithSource } from '../../containers/source';
 
 const basePath = chrome.getBasePath();
 
-// TODO: wire up the date picker to remove the hard-coded start/end dates, which show good data for the KPI event type
-const startDate = 1514782800000;
-const endDate = 1546318799999;
+const HostsTableManage = manageQuery(HostsTable);
+const EventsTableManage = manageQuery(EventsTable);
 
 export const Hosts = pure(() => (
   <WithSource sourceId="default">
     {({ auditbeatIndicesExist }) =>
       auditbeatIndicesExist || isUndefined(auditbeatIndicesExist) ? (
-        <>
-          <EventsQuery sourceId="default" startDate={startDate} endDate={endDate}>
-            {({ kpiEventType, loading }) => (
-              <TypesBar
-                loading={loading}
-                data={
-                  kpiEventType!.map((i: KpiItem) => ({
-                    x: i.count,
-                    y: i.value,
-                  })) as HorizontalBarChartData[]
-                }
-              />
-            )}
-          </EventsQuery>
-          <HostsQuery sourceId="default" startDate={startDate} endDate={endDate} cursor={null}>
-            {({ hosts, totalCount, loading, pageInfo, loadMore }) => (
-              <HostsTable
-                loading={loading}
-                data={hosts}
-                totalCount={totalCount}
-                hasNextPage={getOr(false, 'hasNextPage', pageInfo)!}
-                nextCursor={getOr(null, 'endCursor.value', pageInfo)!}
-                loadMore={loadMore}
-              />
-            )}
-          </HostsQuery>
-          <EventsQuery sourceId="default" startDate={startDate} endDate={endDate}>
-            {({ events, loading }) => (
-              <EventsTable
-                data={events!}
-                loading={loading}
-                startDate={startDate}
-                endDate={endDate}
-              />
-            )}
-          </EventsQuery>
-        </>
+        <GlobalTime>
+          {({ poll, to, from, setQuery }) => (
+            <>
+              <EventsQuery sourceId="default" startDate={to} endDate={from}>
+                {({ kpiEventType, loading }) => (
+                  <TypesBar
+                    loading={loading}
+                    data={
+                      kpiEventType!.map((i: KpiItem) => ({
+                        x: i.count,
+                        y: i.value,
+                      })) as HorizontalBarChartData[]
```

Comment:
You should be fine removing this cast as I don't think it is needed. I do not think it adds additional safety but down the road could mask a subtle TypeScript change of types because it's doing a dynamic cast. When possible, I think removing `as` casts is always better in favor of stricter type checks.

Reply:
it went bye bye when I merged with you PR